### ISSUE TITLE
How to backpropagate the gradient to the index of the superpixel center sample？

### ISSUE THREAD

**Issue by rookiie** [zh460045050/LNSNet#12]
Hi Author, I used the following code to simulate the gradient back-propagation of Eq.3 in the paper and the result shows that it can not back-propagate. The paper and the code do not provide enough information about this detail, so I hope the author will give more explanation.

```java
import torch
import torch.nn as nn

class A(nn.Module):
    def __init__(self):
        super(A, self).__init__()
        self.conv = nn.Linear(10,1)
        self.sigmoid = nn.Sigmoid()    
    
    def forward(self, x):
        ind = self.sigmoid(self.conv(x)) * 9
        ind = torch.floor(ind).long()
        out = x[ind]
        return out
    
net = A().double()
inputs = torch.rand(10,dtype=torch.double, requires_grad=True)
out = net(inputs)
out.sum().backward()

print(net.conv.weight.grad) # output: None
```
